components/tidb_query_executors/src/runner.rs

```diff
@@ -439,15 +433,15 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
     /// with IntervalRange like (k1, k4).
     pub async fn handle_request(&mut self) -> Result<(SelectResponse, Option<IntervalRange>)> {
         let mut chunks = vec![];
-        let mut batch_size = Self::batch_initial_size();
+        let mut batch_size = batch_initial_size();
         let mut warnings = self.config.new_eval_warnings();
         let mut ctx = EvalContext::new(self.config.clone());
         let mut record_all = 0;
 
         loop {
             let mut chunk = Chunk::default();
             let mut sample = self.quota_limiter.new_sample(true);
-            let (drained, record_len) = {
+            let (exec_duration, (drained, record_len)) = {
```

Comment:
Well, it seems more tricky to use the cpu time from `observe_cpu_async`.

First, it may be better to use wall clock time instead of thread cpu time, due to the same reason in https://github.com/tikv/tikv/pull/13314#discussion_r949900731.

And also, the sampler doesn't always record the thread time because it is expensive to get thread CPU time. It depends on whether there is a quota limit. So, you may get zero with the default config.

Reply:
Oh, yes, if the cpu limit is not enabled, the cpu time is not actually observed by the quota sample. I will switch back to use wall clock though it's a bit weird to use it in an async context.

Reply:
Seems there is no good way the track the cpu time used in the async context, and since we already moved the reschedule check to a lower level, the batch size adjust here should be less effective. I will rollback it to the original logic.

Reply:
Let's remove `The "Adjust coprocessor batch-size based on the cost time of each round." ` from the PR description. We can add it back in later PRs if we find some good way.

Reply:
done.